assets/js/atomic/components/product/button/index.js

```diff
@@ -106,6 +106,18 @@ const ProductButton = ( { product, className } ) => {
 		}
 		return productCartDetails.text;
 	};
+
+	// This is a hack to trigger cart updates till we migrate to block based card
+	// that relies on the store, see
+	// https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/1247
+	useEffect( () => {
+		// eslint-disable-next-line no-undef
+		const event = new Event( 'wc_fragment_refresh', {
```

Comment:
Two concerns I have:

- `Event` constructor is not supported by IE 11 (but that might be handled by babel transpile)
- We've introduced a tight dependency here to the browser environment (which can be iterated on if needed separately - and documentation helps for explaining the purpose).

I would suggest though that there be some default assignments outside of the component (which would also remove the necessity for the lint rule exclusion), eg:

```js
const Event = window.Event || {};
```

Reply:
>Event constructor is not supported by IE 11 (but that might be handled by babel transpile)

we can test for `Event` existence and work accordingly.

>We've introduced a tight dependency here to the browser environment

I thought of the same thing, we can iterate on this in the future.

> `const Event = window.Event || {};`
Any examples of this? (defining things outisde component context)